tests/solver.rs

```diff
@@ -362,7 +409,15 @@ fn solve_unsat(provider: BundleBoxProvider, specs: &[&str]) -> String {
 }
 
 /// Solve the problem and returns either a solution represented as a string or an error string.
-fn solve_snapshot(provider: BundleBoxProvider, specs: &[&str]) -> String {
+fn solve_snapshot(mut provider: BundleBoxProvider, specs: &[&str]) -> String {
+    // The test dependency provider uses tokio for sleeping
+    let executor = tokio::runtime::Builder::new_multi_thread()
+        .enable_time()
+        .build()
+        .unwrap();
+
+    provider.runtime = Some(executor);
```

Comment:
This code means that all tests using `solve_snapshot` are now also using the async provider behind the scenes. If that's undesirable, let me know!